src/components/MoneyReportHeader.tsx

```diff
@@ -373,21 +317,226 @@ function MoneyReportHeader({policy, report: moneyRequestReport, transactionThrea
+            />
+        ),
+        [CONST.REPORT.PRIMARY_ACTIONS.PAY]: (
+            <SettlementButton
+                onlyShowPayElsewhere={onlyShowPayElsewhere}
+                currency={moneyRequestReport?.currency}
+                policyID={moneyRequestReport?.policyID}
+                chatReportID={chatReport?.reportID}
+                iouReport={moneyRequestReport}
+                onPress={confirmPayment}
+                enablePaymentsRoute={ROUTES.ENABLE_PAYMENTS}
+                addBankAccountRoute={bankAccountRoute}
+                shouldHidePaymentOptions={!shouldShowPayButton}
+                shouldDisableApproveButton={shouldDisableApproveButton}
+                formattedAmount={!hasOnlyHeldExpenses ? displayedAmount : ''}
+                isDisabled={isOffline && !canAllowSettlement}
+                isLoading={!isOffline && !canAllowSettlement}
+            />
+        ),
+        [CONST.REPORT.PRIMARY_ACTIONS.EXPORT_TO_ACCOUNTING]: (
+            <Button
+                success
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                text={translate('workspace.common.exportIntegrationSelected', {connectionName: connectedIntegration!})}
+                onPress={() => {
+                    if (!connectedIntegration || !moneyRequestReport) {
+                        return;
+                    }
+                    exportToIntegration(moneyRequestReport.reportID, connectedIntegration);
+                }}
+            />
+        ),
+        [CONST.REPORT.PRIMARY_ACTIONS.REMOVE_HOLD]: (
+            <Button
+                success
+                text={translate('iou.unhold')}
+                onPress={() => {
+                    const parentReportAction = getReportAction(moneyRequestReport?.parentReportID, moneyRequestReport?.parentReportActionID);
+
+                    const moneyRequestAction = transactionThreadReportID ? requestParentReportAction : parentReportAction;
+                    if (!moneyRequestAction) {
+                        return;
+                    }
+
+                    changeMoneyRequestHoldStatus(moneyRequestAction);
+                }}
+            />
+        ),
+        [CONST.REPORT.PRIMARY_ACTIONS.MARK_AS_CASH]: (
+            <Button
+                success
+                text={translate('iou.markAsCash')}
+                onPress={markAsCash}
+            />
+        ),
+    };
+
+    const secondaryActions = useMemo(() => {
+        if (!moneyRequestReport) {
+            return [];
         }
+        return getSecondaryReportActions(moneyRequestReport, transactions, violations, policy);
+    }, [moneyRequestReport, policy, transactions, violations]);
+
+    const secondaryActionsImpl: Record<ValueOf<typeof CONST.REPORT.SECONDARY_ACTIONS>, DropdownOption<ValueOf<typeof CONST.REPORT.SECONDARY_ACTIONS>>> = {
+        [CONST.REPORT.SECONDARY_ACTIONS.VIEW_DETAILS]: {
+            value: CONST.REPORT.SECONDARY_ACTIONS.VIEW_DETAILS,
```

Comment:
may be outside this PR, but how do we ensure the order of the menu items matches the list in the design doc?

Reply:
They are returned in correct order by `getSecondaryReportActions`